freezeyt/freezer.py

```diff
@@ -251,38 +189,27 @@ def __init__(self, app, config):
 
             if isinstance(app_config, str):
                 # config file/variable or command line argument
-                self.app = import_variable_from_module(
+                app = import_variable_from_module(
                     app_config, default_variable_name='app',
                 )
             else:
                 # config variable - app as object
-                self.app = app_config
+                app = app_config
         else:
             if app_config is not None:
                 raise ValueError("Application is specified both as parameter and in configuration")
-            self.app = app
+            app = app
+
+        self.app = Middleware(app, config)
```

Comment:
If I understand it well, we cannot make freeze without middleware, so there is not case where we could lost functionality of checking mimetype.